api/router.js

```diff
@@ -6,31 +6,45 @@ var autoredraw = require("../api/autoredraw")
 
 module.exports = function($window, renderer, pubsub) {
 	var router = coreRouter($window)
+	var identity = function(o){return o}
+	var globalId
+	var routeRequested = false
+	function onbeforeresolve(){routeRequested = true}
+
 	var route = function(root, defaultRoute, routes) {
-		var current = {path: null, component: "div"}
+		var current = {path: null, component: "div", render:identity}
 		var replay = router.defineRoutes(routes, function(payload, args, path, route) {
-			args.path = path, args.route = route
-			if (typeof payload.onmatch === "function") {
-				if (typeof payload.view !== "function") payload.view = function(vnode) {return vnode}
-				var resolve = function(component) {
-					current.path = path, current.component = component
-					renderer.render(root, payload.view(Vnode(component, null, args, undefined, undefined, undefined)))
-				}
-				if (path !== current.path) payload.onmatch(Vnode(payload, null, args, undefined, undefined, undefined), resolve)
-				else resolve(current.component)
+			if (!routeRequested) {
+				renderer.render(root, current.render.call(payload, Vnode(current.component, null, args, undefined, undefined, undefined)))
+				return
+			}
+			routeRequested = false
+			var render = route.render || identity
```

Comment:
This allows one to define a global, default layout for all routes by adding a `m.route.render` function. `RouteResolver.render` takes precedence when present on a given route.